test/Nancy.Tests/Unit/DefaultNancyBootstrapperFixture.cs

```diff
@@ -74,22 +80,32 @@ public void Request_should_be_available_to_request_startup()
         public void Container_should_ignore_specified_assemblies()
         {
             // Given
-            var ass = CSharpCodeProvider
-                .CreateProvider("CSharp")
-                .CompileAssemblyFromSource(
-                    new CompilerParameters
-                    {
-                        GenerateInMemory = true,
-                        GenerateExecutable = false,
-                        IncludeDebugInformation = false,
-                        OutputAssembly = "TestAssembly.dll"
-                    },
-                    new[]
-                    {
-                        "public interface IWillNotBeResolved { int i { get; set; } }",
-                        "public class WillNotBeResolved : IWillNotBeResolved { public int i { get; set; } }"
-                    })
-                .CompiledAssembly;
+            var syntaxTree = CSharpSyntaxTree.ParseText(
+                "public interface IWillNotBeResolved { int i { get; set; } }" +
+                "public class WillNotBeResolved : IWillNotBeResolved { public int i { get; set; } }");
+
+            var mscorlib = MetadataReference.CreateFromFile(typeof(object).GetAssembly().Location);
+            var compilation = CSharpCompilation.Create("MyCompilation",
+                new[] { syntaxTree },
+                new[] { mscorlib },
+                new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+            Assembly ass;
+            using (var memoryStream = new MemoryStream())
+            {
+                var emitResult = compilation.Emit(memoryStream);
+
+                Assert.True(emitResult.Success,
+                    "Compilation failed:" + Environment.NewLine + "  " +
+                    string.Join(Environment.NewLine + "  ", emitResult.Diagnostics));
+
+#if CORE

```

Comment:
Pulled out. I took some initiative deviated from your suggestion. Here was my thinking:

1. I didn't think an extension method was best, especially an `Assembly` exception. `MemoryStream` seemed the only option , and that didn't feel right.
2. There didn't seem to be a need to expose this functionality.

So I moved into an `AssemblyHelpers` instead. The location may not be correct - its sibling files contain only tests, not helpers, but I wasn't sure where to put it, and some other directories had mixtures. 

I realize I'm taking liberties, and am happy to change however you direct.